src/components/Remark.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2020 @paritytech/bridge-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+import Keyring from '@polkadot/keyring';
+import { Codec } from '@polkadot/types/types';
+import { u8aToHex } from '@polkadot/util';
+import React, { useState } from 'react';
+import { Button, Container, Input } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+import { useApiSourcePromiseContext } from '../contexts/ApiPromiseSourceContext';
+import { useApiTargetPromiseContext } from '../contexts/ApiPromiseTargetContext';
+import useLoadingApi from '../hooks/useLoadingApi';
+interface Props {
+  className?: string;
+  targetChain: string;
+}
+
+const Remark = ({ className, targetChain }: Props) => {
+  const { api: sourceApi } = useApiSourcePromiseContext();
+  const { api: targetApi } = useApiTargetPromiseContext();
+
+  const areApiLoading = useLoadingApi();
+  const [remarkInput, setRemarkInput] = useState('0x');
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRemarkInput(event.target.value);
+  };
+
+  async function sendMessageRemark() {
+    const keyring = new Keyring({ type: 'sr25519' });
+    const account = keyring.addFromUri('//Alice');
+
+    const remarkCall = await targetApi.tx.system.remark(remarkInput);
+    const remarkInfo = await sourceApi.tx.system.remark(remarkInput).paymentInfo(account);
+    const weight = remarkInfo.weight.toNumber();
+
+    const call = remarkCall.toU8a();
+
+    const lane_id = new Uint8Array(8);
+    const payload = {
+      call,
+      origin: {
+        SourceAccount: account.addressRaw
+      },
+      spec_version: 1,
+      weight
+    };
+
+    // Ignoring custom types missed for TS for now.
+    // Need to apply: https://polkadot.js.org/docs/api/start/typescript.user
+    // @ts-ignore
+    const payloadType = sourceApi.registry.createType('OutboundPayload', payload);
+    // @ts-ignore
+    const messageFeeType = sourceApi.registry.createType('MessageFeeData', {
+      lane_id: '0x00000000',
+      payload: u8aToHex(payloadType.toU8a())
+    });
+
+    const estimatedFeeCall = await sourceApi.rpc.state.call<Codec>(
```

Comment:
I'd enclose the entire `estimate_message_delivery_and_dispatch_fee` workflow into a separate function, cause it most likely is going to be useful outside of `Remark` component.

Reply:
this is why i mentioned that remark.tsx is kind of testing component, all those functions will not remain in the component logic. My plan is to have the end to end working as it is now and then abstract the code accordingly.